src/main/java/com/google/logsummarizer/textloggers/Log4jParser.java

```diff
@@ -0,0 +1,117 @@
+package com.google.logsummarizer.textloggers;
+
+import com.google.logsummarizer.core.ConcreteOperationInterval;
+import com.google.logsummarizer.core.OperationInterval;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+
+/**
+ * Reads the log of a bunny-enabled application running on Dataflow.
+ * (that's the "worker-stdout" log).
+ */
+public class Log4jParser {
```

Comment:
DataflowParser is used to parse logs generated by the workers and retrieved via GCP's log export feature. Those logs are in JSON format.

Log4jParser is for logs that are saved to a file via log4j directly, without going through Google Dataflow's harnesses. An example would be the output from your client, or output from a non-Dataflow program (since this logging, while it has extra Dataflow-helping hooks, would also work just fine in non-Dataflow programs).